forte/pymodule.py

```diff
@@ -478,10 +478,29 @@ def forte_driver(state_weights_map, scf_info, options, ints, mo_space_info):
     # solver for dynamical correlation from DSRG
     correlation_solver_type = options.get_str('CORRELATION_SOLVER')
     if correlation_solver_type != 'NONE':
+
+        # Grab coupling coefficients
+        # NOTE: Orbitals have to be semicanonicalized already to make sure
+        #       DSRG reads consistent CI coefficients before and after SemiCanonical class.
+        do_grad = options.get_str('DERTYPE') == 'FIRST'
+        if active_space_solver_type == "CAS" and do_grad:
+            # This is OK only when running state-specific calculations
+            state = list(state_map.keys())[0]
+            coupling_coefficients = active_space_solver.coupling_coefficients(state, 3)
+            ci_vectors = active_space_solver.eigen_vectors(state)
+
         dsrg_proc = ProcedureDSRG(active_space_solver, state_weights_map, mo_space_info, ints, options, scf_info)
-        return_en = dsrg_proc.compute_energy()
+
+        if do_grad:
+            dsrg_proc.compute_gradient(coupling_coefficients, ci_vectors)
+        else:
+            return_en = dsrg_proc.compute_energy()
+
         dsrg_proc.print_summary()
         dsrg_proc.push_to_psi4_environment()
+
+        if do_grad:
+            return True
```

Comment:
```suggestion
        return_en = dsrg_proc.compute_energy()
        dsrg_proc.print_summary()
        dsrg_proc.push_to_psi4_environment()

        if do_grad:
            dsrg_proc.compute_gradient(coupling_coefficients, ci_vectors)
```

Reply:
Sorry York, maybe I misunderstood your previous comment, but do you still want the energy computation to exist?

Reply:
Yes, because it is confusing that `compute_gradient` actually does `compute_energy`.